src/App.ts

```diff
@@ -1368,29 +1368,43 @@ function buildSource<IsEnterpriseInstall extends boolean>(
       return (body as SlackCommandMiddlewareArgs['body']).team_id;
     }
 
+    const parseTeamId = (
+      bodyAs:
+      | SlackAction
+      | SlackViewAction
+      | SlackShortcut
+      | KnownOptionsPayloadFromType<OptionsSource>,
+    ): string | undefined => {
+      // When the app is installed using org-wide deployment, team property will be null
+      if (typeof bodyAs.team !== 'undefined' && bodyAs.team !== null) {
+        return bodyAs.team.id;
+      }
+
+      // This is the only place where this function might return undefined
+      return bodyAs.user.team_id;
+    };
+
+    if (type === IncomingEventType.ViewAction) {
+      const bodyAsView = body as SlackViewMiddlewareArgs['body'];
+
+      if (bodyAsView.view.app_installed_team_id) {
```

Comment:
Nice! Can we also add a comment for future eyes about when this condition is met, since it's a little nuanced? Maybe mirroring what Kaz has added in his [PR](https://github.com/slackapi/bolt-python/pull/734/files#diff-1b84316ccd4284cd138866cd5772ceec73edbc47bf9cf432bd6d2f8c5db58e64R77)